src/raftstore/store/worker/compact.rs

```diff
@@ -226,16 +226,32 @@ fn collect_ranges_need_compact(
         }
 
         // Current range doesn't need compacting, save previous range that need compacting.
-        if compact_start.is_some() {
-            ranges_need_compact.push_back((compact_start.unwrap(), compact_end.unwrap()));
-            compact_start = None;
-            compact_end = None;
+        debug_assert!(if compact_start.is_some() {
```

Comment:
I would prefer:

```rust
if compact_start.is_some() {
    debug_assert!(compact_end.is_some());
}
```

which looks to be more clear. The optimizer should be able to eliminate the branch when seeing `debug_assert` is nop in release mode.

Reply:
Sure. I'll change it.

Reply:
This is fixed now.